pkg/ipcache/ipcache.go

```diff
@@ -473,16 +473,42 @@ func (ipc *IPCache) RemoveMetadata(prefix netip.Prefix, resource ipcacheTypes.Re
 // labels with these prefixes, thereby making these prefixes selectable in
 // policy via local ("CIDR") identities.
 //
+// It will return a list of identities corresponding the supplied prefixes.
```

Comment:
I like the idea of preserving the existing ipcache behavior and waiting for the updates to make it *back* to the selector cache.

We could implement this one of several ways; my first thought is to block in policy distillation until every CachedSelector (which performed allocation) has been updated with the identities for those allocations.

This would also work well with the fqdn proxy; rather than having the dns layer be responsible for allocation and the selector cache be responsible for deallocation, we could change it so the dns layer only thinks about IPs, and the selectorcache concerns itself with identities.

Since policy distillation necessarily requires the endpoint lock, it means that the ipcache must **never require endpoint locks** to make progress. Right now, this is not true, but not for any fundamental reasons. There are just some read-only fields behind a mutex that can easily be switched to atomic pointers.

Thoughts?